ls1intum/Artemis
# `Programming exercises`: Delete build log entries during deletion of programming submission

### Checklist
#### General


#### Server

### Motivation and Context
See issue #6089: The deletion of a course sometimes fails because the build log entries belonging to a ProgrammingSubmission are not deleted before the removal of the ProgrammingSubmission.

### Description
Delete build log entries before removing the submission & add a test case for it.

### Steps for Testing
1. Create a course with a programming exercise
2. Submit a failing submission (so that build log entries are saved)
3. Delete the course
4. No error should occur



### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage
